lib/App/Config/Chronicle.pm

```diff
@@ -393,6 +409,396 @@ sub _add_dynamic_setting_info {
+    $self->data_set->{global}->set($_, $pairs->{$_}) foreach keys %$pairs;
+    $self->save_dynamic();
+
+    return 1;
+}
+
+sub _store_objects {
+    my ($self, $key_objs_hash, $date_obj, $optional_chron_args) = @_;
+    $self->_store_objects_in_chron($key_objs_hash, $date_obj, $optional_chron_args);
+    $self->_store_objects_in_cache($key_objs_hash) if $self->local_caching;
+    return 1;
+}
+
+sub _store_objects_in_cache {
+    my ($self, $key_objs_hash) = @_;
+    $self->{$_->key} = $_->value foreach (pairs %$key_objs_hash);
+    return 1;
+}
+
+sub _store_objects_in_chron {
+    my ($self, $key_objs_hash, $date_obj, $optional_chron_args) = @_;
+    my @atomic_write_pairs = pairmap { [$self->setting_namespace, $a, $b] } %$key_objs_hash;
+    $self->chronicle_writer->mset(\@atomic_write_pairs, $date_obj, @$optional_chron_args);
+    return 1;
+}
+
+=head2 get
+
+Takes either an arrayref of keys, gets them atomically, and returns a hashref of key->values,
+including the global_revision under the key '_global_revision'
+If a key has an empty value, it will return with undef.
+
+For convenience a get with just a key string will return the value only.
+
+Example:
+    get(['key1','key2','key3',...]);
+Returns:
+    {'key1' => 'value1', 'key2' => 'value2', 'key3' => 'value3',..., '_global_revision' => '<number>'}
+
+Example:
+    get(['key1']);
+Returns:
+    {'key1' => 'value1', '_global_revision' => '<number>'}
+
+Example:
+    get('key1');
+Returns:
+    'value1'
+
+=cut
+
+sub get {
+    my ($self, $keys) = @_;
+    my $single_get;
+
+    unless (ref $keys) {
+        $keys       = [$keys];
+        $single_get = 1;
+    }
+
+    $self->_key_exists($_) or die "Cannot get with key: $_ | Key must be defined" foreach @$keys;
+
+    push @$keys, '_global_rev' unless $single_get;
+
+    my @result_objs = $self->_retrieve_objects($keys);
+
+    if ($single_get) {
+        return $result_objs[0] ? $result_objs[0]->{data} : $self->get_default($keys->[0]);
+    } else {
+        return {map { $keys->[$_] => $result_objs[$_] ? $result_objs[$_]->{data} : $self->get_default($keys->[$_]) } (0 .. scalar @$keys - 1)};
+    }
+}
+
+sub _retrieve_objects {
+    my ($self, $keys) = @_;
+    return $self->_retrieve_objects_from_cache($keys) if $self->local_caching;
+    return $self->_retrieve_objects_from_chron($keys);
+}
+
+sub _retrieve_objects_from_cache {
+    my ($self, $keys) = @_;
+    return map { $self->{$_} } @$keys;
+}
+
+sub _retrieve_objects_from_chron {
+    my ($self, $keys) = @_;
+    my @atomic_read_pairs = map { [$self->setting_namespace, $_] } @$keys;
+    return $self->chronicle_reader->mget(\@atomic_read_pairs);
+}
+
+=head2 get_history
+
+Retreives a past revision of an app config entry, where $rev is the number of revisions in the past requested.
+If the third argument is set to 1 the result of the query will be cached in Redis. This is useful if a certain
```

Comment:
I'd say simply "if the optional third argument is true" - the code simply tests truth, not equality to 1